src/components/menu/menu.tsx

```diff
@@ -443,13 +423,18 @@ export class Menu {
         this.searchValue = query;
         if (query === '') {
             this.searchResults = null;
+            this.loadingSubItems = false;
 
             return;
         }
 
         this.loadingSubItems = true;
 
-        const result = await this.debouncedSearch(query);
+        const result = await this.searcher(query);
+
+        if (this.searchValue !== query) {
```

Comment:
We set `this.searchValue` to `query` a few lines above, how is this ever supposed to be `true`? The early return from this if-statement also forgets to reset `this.loadingSubItems` to `false`?

Reply:
We're awaiting a search request that could take seconds. Meanwhile, other searches that complete faster could be made, or the search is cleared for some reason (`this.clearSearch()` is called from six different places in this component), for example because the component is rendered with new items.

Reply:
Not setting `this.loadingSubItems` to `false` is intentional, because we're no longer loading the same search. We expect `this.loadingSubItems` to be set to `false` when the other search completes or when search is cleared.